src/main/java/com/snowflake/kafka/connector/Utils.java

```diff
@@ -272,6 +278,17 @@ static boolean enableJVMProxy(Map<String, String> config) {
       System.setProperty(HTTPS_PROXY_HOST, host);
       System.setProperty(HTTPS_PROXY_PORT, port);
 
+      // If the user provided the jvm.nonProxy.hosts configuration then we
+      // will append that to the list provided by the JVM argument
+      // -Dhttp.nonProxyHosts and not override it altogether, if it exists.
+      if (nonProxyHosts != null) {
+        nonProxyHosts =
+            (System.getProperty(HTTP_NON_PROXY_HOSTS) != null)
+                ? System.getProperty(HTTP_NON_PROXY_HOSTS) + "|" + nonProxyHosts
```

Comment:
will there be any issue if there're duplicates?

Reply:
@sfc-gh-tzhang no, it's safe.